### ISSUE TITLE
How to change the version of Ace editor?

### ISSUE THREAD

**Issue by lbbc1117** [josdejong/jsoneditor#351]
As Ace 1.26 fixed some bug that bothers me.

**Comment by josdejong** [comment]
You can pass your own version of Ace as a configuration option, see docs:

https://github.com/josdejong/jsoneditor/blob/master/docs/api.md#configuration-options

To prevent bundling Ace editor twice, you can probably use the minimalist version of jsoneditor (see [Which files do I need?](https://github.com/josdejong/jsoneditor/blob/master/dist/which%20files%20do%20I%20need.md)).

**Comment by lbbc1117** [comment]
Thanks,  it worked! 
But when I use the minimalist version of jsoneditor, Ace raised 2 errors as it could not find mode-json.js & theme-jsoneditor.js. 
And the full version works fine.

**Comment by josdejong** [comment]
> But when I use the minimalist version of jsoneditor, Ace raised 2 errors as it could not find mode-json.js & theme-jsoneditor.js.

These files are bundles in the minimalist version. Maybe you load JSONEditor before Ace instead of vice versa?

**Comment by lbbc1117** [comment]
> Maybe you load JSONEditor before Ace instead of vice versa?

It happens when I call
`editor.setMode('code')`,
in spite of the loading order.

And then jsoneditor calls
`aceEditor.setTheme("ace/theme/jsoneditor");
aceEditor.getSession().setMode("ace/mode/json");`

Then Ace appends 2 script tags to header with dynamic src property for above, and the 2 paths were wrong. They are:
 "theme-jsoneditor.js" & "mode-json.js".

Any suggestion?

**Comment by josdejong** [comment]
Turns out the minimalist version of JSONEditor was not loading the `ace/theme/jsoneditor` that it has embedded, sorry for that. I've fixed that in the `develop` branch, and added a code example and more docs on how to use a custom Ace editor (see [08_custom_ace.html](https://github.com/josdejong/jsoneditor/blob/develop/examples/08_custom_ace.html)).

There are two other Ace plugins that JSONEditor loads: `ace/mode/json` and `ace/ext/searchbox`. You will have to load these plugins yourself beforehand or have them available in the folders of your custom Ace editor, then Ace will load them automatically.
